circleguard/enums.py

```diff
@@ -267,6 +272,52 @@ class Mod():
              V2, TD, # we stop caring about order after this point
              FI, RD, CN ,TP, K1, K2, K3, K4, K5, K6, K7, K8, K9, CO, MR]
 
+    # same as ORDER but with NM included
+    ALL = [NM, EZ, HD, HT, DT, _NC, HR, FL, NF, SD, _PF, RX, AP, SO, AT,
+           V2, TD, FI, RD, CN ,TP, K1, K2, K3, K4, K5, K6, K7, K8, K9, CO, MR]
+
+    def __init__(self, value):
+        if type(value) is int:
+            super().__init__(value)
+        if type(value) is str:
+            super().__init__(Mod._parse_mod_string(value))
+
+    @staticmethod
+    def _parse_mod_string(mod_string):
+        """
+        Creates an integer representation of a mod string made up of two letter
+        mod names ("HDHR", for example).
+
+        Arguments
+        ---------
+        mod_string: str
+            The mod string to represent as an int.
+
+        Returns
+        -------
+        int:
+            The integer representation of the mod string.
+
+        Raises
+        ------
+        ValueError:
+            If mod_string is empty, not of even length, or any of its 2-length
+            substrings do not correspond to a ModCombination in Mod.ALL.
+        """
+        if mod_string == "":
+            raise ValueError("Invalid mod string (cannot be empty)")
+        if len(mod_string) % 2 != 0:
+            raise ValueError(f"Invalid mod string {mod_string} (not of even length)")
+        mod_value = 0
+        for i in range(2, len(mod_string) + 1, 2):
+            single_mod_string = mod_string[i - 2: i]
```

Comment:
```suggestion
        for i in range(2, len(mod_string) - 2, 2):
            single_mod_string = mod_string[i: i + 2]
```
easier to understand imo